### PR TITLE
update quantization doc: add x86 backend as default backend of server inference

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86794

### CODE REVIEW

**Comment by jgong5** [comment]
Since x86 is now the default qengine replacing fbgemm, can we only recommend x86 for server CPUs? We can leave a note that fbgemm is still available but not recommended.

File: docs/source/quantization.rst [hunk 1]
```diff
@@ -742,30 +748,31 @@ Backend/Hardware Support
 
 Today, PyTorch supports the following backends for running quantized operators efficiently:
 
-* x86 CPUs with AVX2 support or higher (without AVX2 some operations have inefficient implementations), via `fbgemm <https://github.com/pytorch/FBGEMM>`_
+* x86 CPUs with AVX2 support or higher (without AVX2 some operations have inefficient implementations), via `x86` to apply the optimization of `fbgemm <https://github.com/pytorch/FBGEMM>`_ and `onednn <https://github.com/oneapi-src/oneDNN>`_ (see the details at `RFC <https://github.com/pytorch/pytorch/issues/83888>`_)
```
Comment (jgong5) [inline docs/source/quantization.rst:751]:
```suggestion
* x86 CPUs with AVX2 support or higher (without AVX2 some operations have inefficient implementations), via `x86` optimized by `fbgemm <https://github.com/pytorch/FBGEMM>`_ and `onednn <https://github.com/oneapi-src/oneDNN>`_ (see the details at `RFC <https://github.com/pytorch/pytorch/issues/83888>`_)
```

**Review by jgong5 — CHANGES_REQUESTED** [review]

**Review by jgong5 — APPROVED** [review]

**Review by kit1980 — APPROVED** [review]

**Comment by kit1980** [comment]
@XiaobingSuper feel free to merge when ready.

**Comment by XiaobingSuper** [comment]
@kit1980 @jerryzh168, code is rebased.

**Review by kit1980 — APPROVED** [review]

**Pull request closed without merging**